libs/langchain/langchain/output_parsers/json.py

```diff
@@ -8,6 +8,30 @@
 from langchain.schema import BaseOutputParser, OutputParserException
 
 
+def replace_new_line(match: re.Match[str]) -> str:
+    value = match.group(2)
+    value = re.sub(r"\n", r"\\n", value)
+    value = re.sub(r"\r", r"\\r", value)
+    value = re.sub(r"\t", r"\\t", value)
+    value = re.sub('"', r"\"", value)
+
+    return match.group(1) + value + match.group(3)
+
+
+def custom_parser(multiline_string: str) -> str:
```

Comment:
Do you mind marking these as private functions?
Could you add a bit more documentation to explain what it means to "handle new lines and other special characters" correctly (i.e., explaining that special characters need to be escape properly, so this is searching for instances of special characters that aren't being escaped inside the `action input` etc...)

Reply:
@eyurtsev sure - updated.

Reply:
@eyurtsev  ping.